src/RestControllers/SMART/SMARTConfigurationController.php

```diff
@@ -0,0 +1,127 @@
+<?php
+
+namespace OpenEMR\RestControllers\SMART;
+
+/**
+ * Handles the creation of the smart-configuration used for SMART apps and complies with
+ * SMART on FHIR Core Capabilities 1.0.0
+ * @see http://hl7.org/fhir/smart-app-launch/conformance/index.html
+ *
+ * @package OpenEMR\RestControllers\SMART
+ * @link      http://www.open-emr.org
+ * @author    Stephen Nielson <stephen@nielson.org>
+ * @copyright Copyright (c) 2020 Stephen Nielson <stephen@nielson.org>
+ * @license   https://github.com/openemr/openemr/blob/master/LICENSE GNU General Public License 3
+ */
+class SMARTConfigurationController
+{
+    /**
+     * @var \OpenEMR\RestControllers\AuthorizationController
+     */
+    private $authServer;
+
+    public function __construct(\OpenEMR\RestControllers\AuthorizationController $authServer)
+    {
+        $this->authServer = $authServer;
+    }
+
+    public function getConfig()
+    {
+        $authServer = $this->authServer;
+        // TODO: should we abstract the innards of the REST controller into its own class
+        // so we don't violate single responsibility principle?
+        $metadataController = new \OpenEMR\RestControllers\FHIR\FhirMetaDataRestController();
+        $statement = $metadataController->getMetaData();
+
+        // TODO: merge these with the OAUTH scopes
+        $scopesSupported = [
+            "openid"
+            , "profile"
+//            , "launch"
+            , "launch/patient"
+            , "patient/*.*"
+//            , "user/*.*"
+//            , "offline_access"
+        ];
+        // create hash dictionary
+        $scopes_dict = array_combine($scopesSupported, $scopesSupported);
+        $restAPIs = $statement->getRest();
+        foreach ($restAPIs as $api) {
+            $resources = $api->getResource();
+            foreach ($resources as $resource) {
+                // annoying that we switch into JSON instead of objects here
+                // violates the least surprise principle...
+                $interactions = $resource['interaction'];
+                $resourceType = $resource['type'];
+                foreach ($interactions as $interaction) {
+                    $scopeRead = $resourceType . ".read";
+                    $scopeWrite = $resourceType . ".write";
+                    switch ($interaction['code']) {
+                        case 'read':
```

Comment:
@bradymiller or @sjpadgett Can one of you tell me what is wrong which the style checker here.  I can't figure out what I'm doing wrong.  The stylechecker says I need a colon here... but I have one so I'm not sure what's up with this.

Reply:
I don't see a styling error for the switch however, you've other errors like one at L-258.
Are you looking at the details on the failed tests. Also, life is much easier if you run code through ci local.
I run from terminal inside openemr root as
php  /c/xampp/htdocs/phpcbf.phar -n --standard=ci/phpcs.xml src/RestControllers/AuthorizationController.php

but there are other ways other use on repository.

Reply:
I've been running the stylechecker locally, been trying to wire it up in phpstorm but it doesn't seem to like it too much.  I was going through and fixing things but couldn't figure out for the life of me the error with the switch so I just pushed it and figured I'd ask y'all what's going on here.

also the local ci doesn't appear to check against the php 7.4 & php 8.0.  I've fixed those issues and will push again if I can figure out how to get the style's to work properly.

Reply:
yep I couldn't get to work with phpstorm either but, i'm not very patient and gave up easy.
Still the style beautifier should catch and fix all those. Don't know about PHP8 but mine has been cool for PHP 7.4.

Reply:
wouldn't get to hung up on styling (we can fix that with a one liner after the PR comes in).
If you happen to be using the easy dev docker environment, then can fix styling via:
```
openemr-cmd pf
```
See here for getting openemr-cmd going: https://github.com/openemr/openemr-devops/tree/master/utilities/openemr-cmd

Reply:
This has been very helpful thank you.